pkg/resourcecollector/persistentvolume.go

```diff
@@ -120,10 +121,14 @@ func (r *ResourceCollector) preparePVResourceForApply(
 	}
 
 	pv.Name = updatedName
-	driverName, err := volume.GetPVDriver(&pv)
-	if err != nil {
-		return false, err
+	var driverName string
+	for _, vol := range vInfo {
+		if vol.RestoreVolume == pv.Name {
+			driverName = vol.DriverName
```

Comment:
If we restore to different storage class, the source and destination driver will be different rt?

Reply:
Pls, Ignore this. In the case of kdmp, it is dynamic provision, So it should be taken care.